draft-yasskin-http-origin-signed-responses.md

```diff
@@ -1054,112 +1054,76 @@ This allows the mitigation against downgrades in {{seccons-downgrades}}, but
 prohibits intermediates from providing a cache of the validity information. We
 could do both with a list of URLs.
 
-## application/http-exchange+cbor format for HTTP/1 compatibility # {#application-http-exchange}
+## application/signed-exchange format # {#application-signed-exchange}
+
+To allow signed exchanges to be the targets of `<link rel=prefetch>` tags, we
+define the  `application/signed-exchange` content type that represents a signed
+HTTP exchange, including request metadata and header fields, response metadata
+and header fields, and a response payload.
+
+This content type consists of the concatenation of the following items:
+
+1. The ASCII characters "sxg1" followed by a 0 byte, to serve as a file
+   signature.
+
+   Note: RFC EDITOR PLEASE DELETE THIS NOTE; The implementation of the final RFC
+   MUST use this file signature, but implementations of drafts MUST NOT use it
+   and MUST use another implementation-specific string beginning with "sxg1-" and
+   ending with a 0 byte instead.
+1. 3 bytes storing a big-endian integer `sigLength`. If this is larger
+   than TBD, parsing MUST fail.
+1. `sigLength` bytes holding the `Signature` header field's value
+   ({{signature-header}}).
+1. 3 bytes storing a big-endian integer `headerLength`. If this is larger than
```

Comment:
WDYT about putting the two lengths first? (I.e. swap this one and the one above.) One less state in your parsing state machine. (First you read a fixed 5 + 3 + 3 bytes, then you check lengths, then you read additional `sigLength + headerLength` bytes.)

Reply:
Sure.